lib/api/LogManagerImpl.cpp

```diff
@@ -807,12 +807,53 @@ namespace MAT_NS_BEGIN
     void LogManagerImpl::SetDataInspector(const std::shared_ptr<IDataInspector>& dataInspector)
     {
         LOCKGUARD(m_dataInspectorGuard);
-        m_dataInspector = dataInspector;
+        if(dataInspector == nullptr)
+        {
+            LOG_WARN("Attempting to set nullptr as DataInspector");
+            return;
+        }
+
+        auto itDataInspector = std::find_if(m_dataInspectors.begin(), m_dataInspectors.end(), [&dataInspector](const std::shared_ptr<IDataInspector>& currentInspector)
+        {
+            return strcmp(typeid(*dataInspector).name(),typeid(*currentInspector).name()) == 0;
```

Comment:
Do we need to strcmp the typeid.name()s to compare the types?

Reply:
comparing by `GetName()` now